src/main/java/org/mondemand/Client.java

```diff
@@ -1232,16 +1273,24 @@ private synchronized void dispatchStatsSamples() {
     try {
       Context[] contexts = this.contexts.values().toArray(new Context[0]);
       StatsMessage[] statsMsgs = this.stats.values().toArray(new StatsMessage[0]);
+
+      // snapshot samples map for dispatch
       SamplesMessage[] samplesMsgs = this.samples.values().toArray(new SamplesMessage[0]);
+      this.samples = new ConcurrentHashMap<String, SamplesMessage>();
```

Comment:
swap out samples map when dispatching... we may miss some samples for increments being made to the snapshot map, but that should be fine.